cmd/registry/main.go

```diff
@@ -17,8 +17,31 @@ import (
 	_ "github.com/docker/distribution/registry/storage/driver/s3-aws"
 	_ "github.com/docker/distribution/registry/storage/driver/s3-goamz"
 	_ "github.com/docker/distribution/registry/storage/driver/swift"
+	"github.com/docker/distribution/version"
+	"github.com/spf13/cobra"
 )
 
+var showVersion bool
+
+func init() {
+	rootCmd.Flags().BoolVarP(&showVersion, "version", "v", false, "show the version and exit")
+}
+
+var rootCmd = &cobra.Command{
```

Comment:
Shouldn't this be in the `registry` package? This way, you don't need to do the command assembly in the main package.